source/slang/syntax.cpp

```diff
@@ -355,19 +355,21 @@ void Type::accept(IValVisitor* visitor, void* extra)
         auto arrType = type->AsArrayType();
         if (!arrType)
             return false;
-        return (ArrayLength == arrType->ArrayLength && baseType->Equals(arrType->baseType.Ptr()));
+        return (ArrayLength->EqualsVal(arrType->ArrayLength) && baseType->Equals(arrType->baseType.Ptr()));
```

Comment:
The two bugs in this file have been around a long time, and I'm kind of amazed they hand't bit us before. Thanks for catching these!